### PR TITLE
Bug 1550098 - Trailhead initial uplift for Firefox 67.0

### PR DESCRIPTION
**Pull request by Mardak**
r?@rlr This is on top of #4953 .. should we just merge that?

There were some conflicts as noted in https://bugzilla.mozilla.org/show_bug.cgi?id=1545802#c1

Also had to fix a <> -> React.Fragment

There were some new conflicts for asrouter changes on master that aren't needed for beta but touched code next to trailhead related changes.

### CODE REVIEW

**Comment by Mardak** [comment]
The trailhead commit includes these commits:
https://github.com/mozilla/activity-stream/compare/c099d845...681b99e6

**Comment by rlr** [comment]
We can merge mine or create a beta-trailhead branch? But I guess if that part needs any change we can put it into the trailhead commit so....

File: content-src/asrouter/asrouter-content.jsx [hunk 1]
```diff
@@ -280,6 +284,21 @@ export class ASRouterUISurface extends React.PureComponent {
...
   }
 
+  renderTrailhead() {
+    const {message} = this.state;
+    if (message.template === "trailhead") {
+      return (<Trailhead
+        document={this.props.document}
+        message={message}
+        onAction={ASRouterUtils.executeAction}
+        onDoneButton={this.dismissBundle(this.state.bundle.bundle)}
```
Comment (rlr) [inline content-src/asrouter/asrouter-content.jsx]: just realized this isn't used in `Trailhead`. Probably fine. Looks like it's to dismiss a bundle of cards but not sure why as they won't appear again anyway.

**Comment by rlr** [comment]
sadly, beta doesn't have the new asrouter admin changes so the special snippet appears at the bottom of the page if you test from there. but that's not worth uplifting

File: bin/render-activity-stream-html.js [hunk 2]
```diff
@@ -104,10 +104,9 @@ for (const src of ${JSON.stringify(scripts, null, 2)}) {
     <link rel="stylesheet" href="${options.baseUrl}css/activity-stream.css" />
   </head>
   <body class="activity-stream">
+    <div id="header-asrouter-container" role="presentation"></div>
     <div id="root">${isPrerendered ? html : "<!-- Regular React Rendering -->"}</div>
-    <div id="snippets-container">
-      <div id="snippets"></div>
-    </div>${options.noscripts ? "" : scriptTag}
+    <div id="footer-asrouter-container" role="presentation"></div>${options.noscripts ? "" : scriptTag}
```
Comment (rlr) [inline bin/render-activity-stream-html.js]: so looking back at this.... I am sure this breaks legacy snippets in 67 but they aren’t officially removed until 68. So we can either change our portal code to use `#snippets-container` or update the snippets code to use this or leave it broken. @piatra thoughts? Do we need legacy snippets in 67?

Comment (rlr) [reply bin/render-activity-stream-html.js]: this seems to work:

```
diff --git a/bin/render-activity-stream-html.js b/bin/render-activity-stream-html.js
index cf55ca6c..d76da095 100644
--- a/bin/render-activity-stream-html.js
+++ b/bin/render-activity-stream-html.js
@@ -106,7 +106,9 @@ for (const src of ${JSON.stringify(scripts, null, 2)}) {
   <body class="activity-stream">
     <div id="header-asrouter-container" role="presentation"></div>
     <div id="root">${isPrerendered ? html : "<!-- Regular React Rendering -->"}</div>
-    <div id="footer-asrouter-container" role="presentation"></div>${options.noscripts ? "" : scriptTag}
+    <div id="snippets-container" role="presentation">
+      <div id="snippets"></div>
+    </div>${options.noscripts ? "" : scriptTag}
   </body>
 </html>
 `;
diff --git a/content-src/asrouter/asrouter-content.jsx b/content-src/asrouter/asrouter-content.jsx
index 5acb4a62..c1c3b600 100644
--- a/content-src/asrouter/asrouter-content.jsx
+++ b/content-src/asrouter/asrouter-content.jsx
@@ -94,7 +94,7 @@ export class ASRouterUISurface extends React.PureComponent {
     this.state = {message: {}, bundle: {}};
     if (props.document) {
       this.headerPortal = props.document.getElementById("header-asrouter-container");
-      this.footerPortal = props.document.getElementById("footer-asrouter-container");
+      this.footerPortal = props.document.getElementById("snippets-container");
     }
   }
```

**Pull request merged**